src/java/org/apache/cassandra/index/sai/disk/IndexSearcherContext.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk;
+
+import org.apache.cassandra.index.sai.SSTableQueryContext;
+import org.apache.cassandra.index.sai.utils.PrimaryKey;
+
+public class IndexSearcherContext
```

Comment:
nit: Won't be super prescriptive, but class naming that reflects we're searching a segment w/ this information might be helpful. (Also probably one of the bits that can move to a `segment` package.)

Reply:
I think we already named this `IndexSegmentSearchContext` above (or I renamed it) but moving it to the `segment` package makes sense.